worlds/sm64ex/Options.py

```diff
@@ -114,35 +115,37 @@ class StrictMoveRequirements(DefaultOnToggle):
     if Move Randomization is enabled"""
     display_name = "Strict Move Requirements"
 
-def getMoveRandomizerOption(action: str):
-    class MoveRandomizerOption(Toggle):
-        """Mario is unable to perform this action until a corresponding item is picked up.
-        This option is incompatible with builds using a 'nomoverando' branch."""
-        display_name = f"Randomize {action}"
-    return MoveRandomizerOption
-
-
-sm64_options: typing.Dict[str, type(Option)] = {
-    "AreaRandomizer": AreaRandomizer,
-    "BuddyChecks": BuddyChecks,
-    "ExclamationBoxes": ExclamationBoxes,
-    "ProgressiveKeys": ProgressiveKeys,
-    "EnableCoinStars": EnableCoinStars,
-    "StrictCapRequirements": StrictCapRequirements,
-    "StrictCannonRequirements": StrictCannonRequirements,
-    "StrictMoveRequirements": StrictMoveRequirements,
-    "AmountOfStars": AmountOfStars,
-    "FirstBowserStarDoorCost": FirstBowserStarDoorCost,
-    "BasementStarDoorCost": BasementStarDoorCost,
-    "SecondFloorStarDoorCost": SecondFloorStarDoorCost,
-    "MIPS1Cost": MIPS1Cost,
-    "MIPS2Cost": MIPS2Cost,
-    "StarsToFinish": StarsToFinish,
-    "death_link": DeathLink,
-    "CompletionType": CompletionType,
-}
-
-for action in action_item_table:
-    # HACK: Disable randomization of double jump
-    if action == 'Double Jump': continue
-    sm64_options[f"MoveRandomizer{action.replace(' ','')}"] = getMoveRandomizerOption(action)
+class EnableMoveRandomizer(Toggle):
+    """Mario is unable to perform some actions until a corresponding item is picked up.
+    This option is incompatible with builds using a 'nomoverando' branch.
+    Specific actions to randomize can be specified in the YAML."""
+    display_name = "Enable Move Randomizer"
+
+class MoveRandomizerActions(OptionSet):
+    """Which actions to randomize when Move Randomizer is enabled"""
```

Comment:
I think it would make sense to include the list of keys in this description. Finding them otherwise is rather complicated.

Reply:
This is already handled by valid_keys and default values.

![image](https://github.com/ArchipelagoMW/Archipelago/assets/89949176/07f7c4a2-ffab-4a27-9004-5250108430c6)

![image](https://github.com/ArchipelagoMW/Archipelago/assets/89949176/7421eee0-7b21-45fe-b1f9-97b47c526948)

Reply:
That only appears if you're using that specific part of the website. If you're editing a yaml directly or using the options page, that does not work

Reply:
> That only appears if you're using that specific part of the website (the weighted options page). If you're editing a yaml directly or using the regular options page, that does not work

In what circumstance would a user be able to see the description but not the list?  The option doesn't appear at all in the regular options page (like all OptionSet options), the template yaml shows all of the items by default, and exported options also show all the items by default.

Reply:
A user has no reason to assume that that list is the full list. For an example: I went to edit yamls to test for this and saw the OptionSet listed out with the default, but I had no way of knowing if the default actually contained every key. I had to go check the `options.py` file to ensure that this was the case. Alternatively: If somebody shares a yaml where they removed some of the keys, somebody cannot edit that yaml to add more keys without guessing